### ISSUE TITLE
Crash when deleting and recreating files

### ISSUE THREAD

**Issue by samtygier** [nextcloud/desktop#1176]
When running `make` and `make clean` in a software project in nextcloud sync'd folder i get a crash. I can reproduce with a simple script that creates and deletes a folder full of small files.

```
test_dir="/home/sam/Nextcloud/test/files"

for val in aaa bbb ccc ddd eee fff ggg hhh
do
        rm -rf $test_dir
        sleep 5
        mkdir -p $test_dir

        for n in `seq 1000`
        do
                echo $val $n > ${test_dir}/file_${n}.txt
        done
done
```

### Expected behaviour
Not crash

### Actual behaviour
Crash

```
Thread 1 "nextcloud" received signal SIGSEGV, Segmentation fault.
0x00007ffff7daac46 in OCC::PropagatorCompositeJob::scheduleSelfOrChild() () from /lib64/libnextcloudsync.so.0

(gdb) bt full
#0  0x00007ffff7daac46 in OCC::PropagatorCompositeJob::scheduleSelfOrChild() () from /lib64/libnextcloudsync.so.0
No symbol table info available.
#1  0x00007ffff7dab383 in OCC::PropagateDirectory::scheduleSelfOrChild() () from /lib64/libnextcloudsync.so.0
No symbol table info available.
#2  0x00007ffff7daabfd in OCC::PropagatorCompositeJob::scheduleSelfOrChild() () from /lib64/libnextcloudsync.so.0
No symbol table info available.
#3  0x00007ffff7dab383 in OCC::PropagateDirectory::scheduleSelfOrChild() () from /lib64/libnextcloudsync.so.0
No symbol table info available.
#4  0x00007ffff7daabfd in OCC::PropagatorCompositeJob::scheduleSelfOrChild() () from /lib64/libnextcloudsync.so.0
No symbol table info available.
#5  0x00007ffff7dab383 in OCC::PropagateDirectory::scheduleSelfOrChild() () from /lib64/libnextcloudsync.so.0
No symbol table info available.
#6  0x00007ffff7da6c29 in OCC::OwncloudPropagator::scheduleNextJobImpl() () from /lib64/libnextcloudsync.so.0
No symbol table info available.
#7  0x00007fffed7559b6 in QtPrivate::QSlotObjectBase::call (a=0x7fffffffd8a0, r=<optimized out>, this=<optimized out>)
    at ../../include/QtCore/../../src/corelib/kernel/qobjectdefs_impl.h:376
No locals.
#8  QSingleShotTimer::timerEvent (this=0x55555638db00) at kernel/qtimer.cpp:318
        args = {0x0}
#9  0x00007fffed74a0a5 in QObject::event (this=0x55555638db00, e=<optimized out>) at kernel/qobject.cpp:1273
No locals.
#10 0x00007fffee02da66 in QApplicationPrivate::notify_helper (this=this@entry=0x55555586d560, receiver=receiver@entry=0x55555638db00, 
    e=e@entry=0x7fffffffdb40) at kernel/qapplication.cpp:3726
        consumed = <optimized out>
#11 0x00007fffee036ec0 in QApplication::notify (this=0x7fffffffde70, receiver=0x55555638db00, e=0x7fffffffdb40)
    at kernel/qapplication.cpp:3485
        w = <optimized out>
        extra = <optimized out>
        isProxyWidget = <optimized out>
        d = <optimized out>
        res = false
        me = <optimized out>
#12 0x00007fffed71f2f8 in QCoreApplication::notifyInternal2 (receiver=0x55555638db00, event=event@entry=0x7fffffffdb40)
    at kernel/qcoreapplication.cpp:1047
        selfRequired = true
        result = false
        cbdata = {0x55555638db00, 0x7fffffffdb40, 0x7fffffffdacf}
        d = <optimized out>
        threadData = 0x555555855580
        scopeLevelCounter = {threadData = 0x555555855580}
#13 0x00007fffed77153b in QCoreApplication::sendEvent (event=0x7fffffffdb40, receiver=<optimized out>)
    at ../../include/QtCore/../../src/corelib/kernel/qcoreapplication.h:234
No locals.
#14 QTimerInfoList::activateTimers (this=0x55555598f8d0) at kernel/qtimerinfo_unix.cpp:643
        e = {<QEvent> = {_vptr.QEvent = 0x7fffed9aacb0 <vtable for QTimerEvent+16>, static staticMetaObject = {d = {superdata = 0x0, 
                stringdata = 0x7fffed89d3c0 <qt_meta_stringdata_QEvent>, data = 0x7fffed89ce00 <qt_meta_data_QEvent>, 
                static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}, d = 0x0, t = 1, posted = 0, spont = 0, 
            m_accept = 1, reserved = 2757}, id = 16}
        currentTimerInfo = 0x0
        n_act = 0
        maxCount = 10
        currentTime = {tv_sec = 18785, tv_nsec = 175658175}
#15 0x00007fffed771e34 in timerSourceDispatch (source=<optimized out>) at kernel/qeventdispatcher_glib.cpp:182
        timerSource = <optimized out>
        timerSource = <optimized out>
#16 idleTimerSourceDispatch (source=source@entry=0x55555598f9d0) at kernel/qeventdispatcher_glib.cpp:229
        timerSource = <optimized out>
#17 0x00007fffebf9afa0 in g_main_dispatch (context=0x7fffd4004fd0) at ../glib/gmain.c:3189
        dispatch = <optimized out>
        prev_source = 0x0
        was_in_call = 0
        user_data = 0x0
        callback = 0x0
        cb_funcs = 0x0
        cb_data = 0x0
        need_destroy = <optimized out>
        source = 0x55555598f9d0
        current = 0x555555c2eb70
        i = 1
        __FUNCTION__ = "g_main_dispatch"
#18 g_main_context_dispatch (context=context@entry=0x7fffd4004fd0) at ../glib/gmain.c:3854
No locals.
#19 0x00007fffebf9b338 in g_main_context_iterate (context=context@entry=0x7fffd4004fd0, block=block@entry=1, dispatch=dispatch@entry=1, 
    self=<optimized out>) at ../glib/gmain.c:3927
        max_priority = 0
        timeout = 0
        some_ready = 1
        nfds = <optimized out>
        allocated_nfds = 8
        fds = 0x5555562af540
#20 0x00007fffebf9b3e3 in g_main_context_iteration (context=0x7fffd4004fd0, may_block=may_block@entry=1) at ../glib/gmain.c:3988
        retval = <optimized out>
#21 0x00007fffed772242 in QEventDispatcherGlib::processEvents (this=0x55555598f760, flags=...) at kernel/qeventdispatcher_glib.cpp:422
        d = 0x555555987380
        canWait = true
        savedFlags = {i = 0}
        result = <optimized out>
#22 0x00007fffd8bb4b35 in QPAEventDispatcherGlib::processEvents (this=0x55555598f760, flags=...) at qeventdispatcher_glib.cpp:69
        didSendEvents = <optimized out>
#23 0x00007fffed71e2eb in QEventLoop::exec (this=this@entry=0x7fffffffddb0, flags=..., flags@entry=...)
    at ../../include/QtCore/../../src/corelib/global/qflags.h:140
        d = 0x555556296140
        locker = {val = 93824995384128}
        ref = {d = 0x555556296140, locker = @0x7fffffffdd38, exceptionCaught = true}
        app = <optimized out>
#24 0x00007fffed725fb6 in QCoreApplication::exec () at ../../include/QtCore/../../src/corelib/global/qflags.h:120
        threadData = 0x555555855580
        eventLoop = {<QObject> = {_vptr.QObject = 0x7fffed9aaa48 <vtable for QEventLoop+16>, static staticMetaObject = {d = {
                superdata = 0x0, stringdata = 0x7fffed8a2020 <qt_meta_stringdata_QObject>, 
                data = 0x7fffed8a1f00 <qt_meta_data_QObject>, 
                static_metacall = 0x7fffed751380 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x555556296140}, static staticQtMetaObject = {d = {
                superdata = 0x0, stringdata = 0x7fffed8a4de0 <qt_meta_stringdata_Qt>, data = 0x7fffed8a2140 <qt_meta_data_Qt>, 
                static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
              superdata = 0x7fffed9a30c0 <QObject::staticMetaObject>, stringdata = 0x7fffed89c4e0 <qt_meta_stringdata_QEventLoop>, 
              data = 0x7fffed89c480 <qt_meta_data_QEventLoop>, 
              static_metacall = 0x7fffed71e000 <QEventLoop::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
              relatedMetaObjects = 0x0, extradata = 0x0}}}
        returnCode = <optimized out>
#25 0x00007fffedabec10 in QGuiApplication::exec () at kernel/qguiapplication.cpp:1762
No locals.
#26 0x00007fffee02d9d9 in QApplication::exec () at kernel/qapplication.cpp:2900
No locals.
#27 0x0000555555612c7f in main (argc=<optimized out>, argv=<optimized out>)
    at /usr/src/debug/nextcloud-client-2.5.1-2.fc30.x86_64/src/gui/main.cpp:167
        app = {<SharedTools::QtSingleApplication> = {<QApplication> = {<QGuiApplication> = {<QCoreApplication> = {<QObject> = {
                    _vptr.QObject = 0x55555583c7c8 <vtable for OCC::Application+16>, static staticMetaObject = {d = {superdata = 0x0, 
                        stringdata = 0x7fffed8a2020 <qt_meta_stringdata_QObject>, data = 0x7fffed8a1f00 <qt_meta_data_QObject>, 
                        static_metacall = 0x7fffed751380 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                        relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x55555586d560}, static staticQtMetaObject = {d = {
                        superdata = 0x0, stringdata = 0x7fffed8a4de0 <qt_meta_stringdata_Qt>, data = 0x7fffed8a2140 <qt_meta_data_Qt>, 
                        static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                      superdata = 0x7fffed9a30c0 <QObject::staticMetaObject>, 
                      stringdata = 0x7fffed89cb20 <qt_meta_stringdata_QCoreApplication>, 
                      data = 0x7fffed89ca00 <qt_meta_data_QCoreApplication>, 
                      static_metacall = 0x7fffed720d10 <QCoreApplication::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, static self = 0x7fffffffde70}, static staticMetaObject = {d = {
                    superdata = 0x7fffed9aabe0 <QCoreApplication::staticMetaObject>, 
                    stringdata = 0x7fffedddae00 <qt_meta_stringdata_QGuiApplication>, 
                    data = 0x7fffedddab80 <qt_meta_data_QGuiApplication>, 
                    static_metacall = 0x7fffedacada0 <QGuiApplication::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                    relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                  superdata = 0x7fffedeaaf40 <QGuiApplication::staticMetaObject>, 
                  stringdata = 0x7fffee3cda00 <qt_meta_stringdata_QApplication>, data = 0x7fffee3cd880 <qt_meta_data_QApplication>, 
                  static_metacall = 0x7fffee034290 <QApplication::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                  relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                superdata = 0x7fffee5234e0 <QApplication::staticMetaObject>, 
                stringdata = 0x555555757980 <qt_meta_stringdata_SharedTools__QtSingleApplication>, 
                data = 0x555555757860 <qt_meta_data_SharedTools__QtSingleApplication>, 
                static_metacall = 0x55555572a3c0 <SharedTools::QtSingleApplication::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, firstPeer = -1, instances = 0x555555cd7470, pidPeer = 0x555555d0f920, actWin = 
    0x0, appId = {static null = {<No data fields>}, d = 0x55555586d520}, block = false}, static staticMetaObject = {d = {
              superdata = 0x555555843900 <SharedTools::QtSingleApplication::staticMetaObject>, 
              stringdata = 0x555555761920 <qt_meta_stringdata_OCC__Application>, data = 0x555555761780 <qt_meta_data_OCC__Application>, 
              static_metacall = 0x55555572bf50 <OCC::Application::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
              relatedMetaObjects = 0x0, extradata = 0x0}}, _gui = {wp = {d = 0x5555561aaea0, value = 0x55555598b410}}, 
          _theme = 0x55555586d4f0, _helpOnly = false, _versionOnly = false, _startedAt = {t1 = 18463, t2 = 768252412}, 
          _showLogWindow = false, _logFile = {static null = {<No data fields>}, d = 0x7fffed7d9b40 <QArrayData::shared_null>}, 
          _logDir = {static null = {<No data fields>}, d = 0x7fffed7d9b40 <QArrayData::shared_null>}, _logExpire = 0, 
          _logFlush = false, _logDebug = false, _userTriggeredConnect = false, _debugMode = false, _proxy = {<QObject> = {
              _vptr.QObject = 0x55555583ca70 <vtable for OCC::ClientProxy+16>, static staticMetaObject = {d = {superdata = 0x0, 
                  stringdata = 0x7fffed8a2020 <qt_meta_stringdata_QObject>, data = 0x7fffed8a1f00 <qt_meta_data_QObject>, 
                  static_metacall = 0x7fffed751380 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                  relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x555555c9a010}, static staticQtMetaObject = {d = {
                  superdata = 0x0, stringdata = 0x7fffed8a4de0 <qt_meta_stringdata_Qt>, data = 0x7fffed8a2140 <qt_meta_data_Qt>, 
                  static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                superdata = 0x7fffed9a30c0 <QObject::staticMetaObject>, 
                stringdata = 0x555555761680 <qt_meta_stringdata_OCC__ClientProxy>, 
                data = 0x555555761620 <qt_meta_data_OCC__ClientProxy>, 
                static_metacall = 0x555555727b00 <OCC::ClientProxy::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                relatedMetaObjects = 0x0, extradata = 0x0}}}, _networkConfigurationManager = {<QObject> = {
              _vptr.QObject = 0x7fffee6f6338 <vtable for QNetworkConfigurationManager+16>, static staticMetaObject = {d = {
                  superdata = 0x0, stringdata = 0x7fffed8a2020 <qt_meta_stringdata_QObject>, 
                  data = 0x7fffed8a1f00 <qt_meta_data_QObject>, 
                  static_metacall = 0x7fffed751380 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                  relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x555555d106e0}, static staticQtMetaObject = {d = {
                  superdata = 0x0, stringdata = 0x7fffed8a4de0 <qt_meta_stringdata_Qt>, data = 0x7fffed8a2140 <qt_meta_data_Qt>, 
                  static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                superdata = 0x7fffed9a30c0 <QObject::staticMetaObject>, 
                stringdata = 0x7fffee6b1540 <qt_meta_stringdata_QNetworkConfigurationManager>, 
                data = 0x7fffee6b1440 <qt_meta_data_QNetworkConfigurationManager>, 
                static_metacall = 0x7fffee615890 <QNetworkConfigurationManager::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}, _checkConnectionTimer = {<QObject> = {
              _vptr.QObject = 0x7fffed9ab3c0 <vtable for QTimer+16>, static staticMetaObject = {d = {superdata = 0x0, 
                  stringdata = 0x7fffed8a2020 <qt_meta_stringdata_QObject>, data = 0x7fffed8a1f00 <qt_meta_data_QObject>, 
                  static_metacall = 0x7fffed751380 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                  relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x555555c8d720}, static staticQtMetaObject = {d = {
                  superdata = 0x0, stringdata = 0x7fffed8a4de0 <qt_meta_stringdata_Qt>, data = 0x7fffed8a2140 <qt_meta_data_Qt>, 
                  static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                superdata = 0x7fffed9a30c0 <QObject::staticMetaObject>, stringdata = 0x7fffed8b0dc0 <qt_meta_stringdata_QTimer>, 
                data = 0x7fffed8b0ce0 <qt_meta_data_QTimer>, 
                static_metacall = 0x7fffed755ce0 <QTimer::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                relatedMetaObjects = 0x0, extradata = 0x0}}, id = 14, inter = 32000, del = 0, single = 0, nulltimer = 0, type = 1}, 
          _folderManager = {d = 0x555555caf440}}
        __PRETTY_FUNCTION__ = "int main(int, char**)"
        updater = <optimized out>
```

### Steps to reproduce
1. Run the above script to delete and create files in the nextcloud folder
2. repeat until it crashes

### Client configuration
Client version: nextcloud-client-2.5.1-2.fc30.x86_64

Operating system: Fedora Linux

Qt version used by client package (Linux only, see also Settings dialog):
Built from Git revision b37cbe on Feb 2 2019, 01:05:06 using Qt 5.11.3, OpenSSL 1.1.1b FIPS 26 Feb 2019

Client package (From Nextcloud or distro) (Linux only): Fedora package

Installation path of client: /usr/bin/nextcloud

### Server configuration

Snap package 15.0.5snap2

Operating system: Fedora

Web server: Apache

Database: mysql

PHP version: 

Nextcloud version:15.0.5snap2

Storage backend (external storage):

### Logs

```[OCC::ActivityListModel::addErrorToActivityList 	Error successfully added to the notification list:  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_600.txt"
[OCC::PropagateItemJob::done 	Could not complete propagation of "test/files/file_601.txt" by OCC::PropagateUploadFileV1(0x56165eacc7d0) with status 3 and error: "File Removed (start upload) /home/sam/Nextcloud/test/files/file_601.txt"
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_601.txt"  retrieved resulted in  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_601.txt"
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_601.txt"  retrieved resulted in error  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_601.txt"
[OCC::ActivityListModel::addErrorToActivityList 	Error successfully added to the notification list:  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_601.txt"
[OCC::PropagateItemJob::done 	Could not complete propagation of "test/files/file_602.txt" by OCC::PropagateUploadFileV1(0x56165ea17350) with status 3 and error: "File Removed (start upload) /home/sam/Nextcloud/test/files/file_602.txt"
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_602.txt"  retrieved resulted in  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_602.txt"
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_602.txt"  retrieved resulted in error  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_602.txt"
[OCC::ActivityListModel::addErrorToActivityList 	Error successfully added to the notification list:  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_602.txt"
[OCC::PropagateItemJob::done 	Could not complete propagation of "test/files/file_603.txt" by OCC::PropagateUploadFileV1(0x56165ea07b00) with status 3 and error: "Local file changed during syncing. It will be resumed."
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_603.txt"  retrieved resulted in  "Local file changed during syncing. It will be resumed."
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_603.txt"  retrieved resulted in error  "Local file changed during syncing. It will be resumed."
[OCC::ActivityListModel::addErrorToActivityList 	Error successfully added to the notification list:  "Local file changed during syncing. It will be resumed."
[OCC::PropagateItemJob::done 	Could not complete propagation of "test/files/file_603.txt" by OCC::PropagateUploadFileV1(0x56165ea07b00) with status 3 and error: "Local file changed during syncing. It will be resumed."
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_603.txt"  retrieved resulted in  "Local file changed during syncing. It will be resumed."
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_603.txt"  retrieved resulted in error  "Local file changed during syncing. It will be resumed."
[OCC::ActivityListModel::addErrorToActivityList 	Error successfully added to the notification list:  "Local file changed during syncing. It will be resumed."
[OCC::PropagatorCompositeJob::slotSubJobFinished 	ASSERT: "i >= 0" in file /builddir/build/BUILD/desktop-2.5.1/src/libsync/owncloudpropagator.cpp, line 834
[OCC::PropagateItemJob::done 	Could not complete propagation of "test/files/file_604.txt" by OCC::PropagateUploadFileV1(0x56165eab9860) with status 3 and error: "File Removed (start upload) /home/sam/Nextcloud/test/files/file_604.txt"
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_604.txt"  retrieved resulted in  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_604.txt"
[OCC::ActivityWidget::slotItemCompleted 	Item  "test/files/file_604.txt"  retrieved resulted in error  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_604.txt"
[OCC::ActivityListModel::addErrorToActivityList 	Error successfully added to the notification list:  "File Removed (start upload) /home/sam/Nextcloud/test/files/file_604.txt"
```

**Comment by mreppen** [comment]
I believe I am experiencing this problem as well. The client often crashes when compiling LaTeX documents, which includes generation and deletion of files in quick succession.

**Comment by DominiqueFuchs** [comment]
The log files indicate that the issue discussed here was resolved by #1427

In case you're still struggling with this issue feel free to try the daily build (fix is merged in master, not yet in a stable release - following soon)

**Comment by samtygier** [comment]
I can't trigger this in 2.6.1. Thanks.
